sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1203,6 +1207,32 @@ def check_datapath_state(self, api, channel, states):
 
         return done
 
+    def get_configured_laser_freq(self, lport):
+        """
+           Return the Tx power configured by user in CONFIG_DB's PORT table
+        """
+        freq = 0
+        asic_index = self.port_mapping.get_asic_id_for_logical_port(lport)
+        port_tbl = self.xcvr_table_helper.get_cfg_port_tbl(asic_index)
+
+        found, port_info = port_tbl.get(lport)
+        if found and 'laser_freq' in dict(port_info):
+            freq = dict(port_info)['laser_freq']
+        return int(freq)
+
+    def get_configured_tx_power(self, lport):
+        """
+           Return the Tx power configured by user in CONFIG_DB's PORT table
+        """
+        power = 0
+        asic_index = self.port_mapping.get_asic_id_for_logical_port(lport)
+        port_tbl = self.xcvr_table_helper.get_cfg_port_tbl(asic_index)
+
+        found, port_info = port_tbl.get(lport)
+        if found and 'tx_power' in dict(port_info):
+            power = dict(port_info)['tx_power']
```

Comment:
Similarly. What if found is True but no tx power is configured on the port. Should we enable a default power setting?

Reply:
The default is the module's default. This API gets the value explicitly configured by the user

Reply:
Similar to the previous comment, Can we test this feature with a ZR module by erasing the port's tx power config and see how the module will be configured.

Reply:
If tx power is not configured, module comes with module's default Tx power.